### PR TITLE
Add getAggregatedAttestationV2 to support EIP-7549

### LINKED ISSUE #445: Update attestation related APIs to be compatible with EIP-7549
attestations for `Electra` require an additional `committee_bits` field.

I think these APIs are effected.
`/eth/v1/beacon/pool/attestations`
`/eth/v1/beacon/blocks/{block_id}/attestations`
`/eth/v1/validator/aggregate_attestation`

### PR DESCRIPTION
**Pull request by ensi321**
`AttestationData` post-electra is committee-indifferent, and thus, the same `attestation_data_root` might be shared across different committees. 

An aggregator is only interested in attestations in its corresponding committee. `getAggregatedAttestationV1` might aggregate attestations from multiple committees since `committee_index` is not in its params. 

This PR adds `getAggregatedAttestationV2` to aggregate attestations given `attestation_data_root` + `slot` + `committee_index`.

Part of #445

### CODE REVIEW

File: apis/validator/aggregate_attestation.v2.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      description: "Returns aggregated `Attestation` object with same `AttestationData` root, slot and committee index."
+      content:
+        application/json:
+          schema:
+            title: GetAggregatedAttestationV2Response
+            type: object
+            required: [data]
+            properties:
+              data:
+                $ref: '../../beacon-node-oapi.yaml#/components/schemas/Electra.Attestation'
```
Comment (nflaig) [inline apis/validator/aggregate_attestation.v2.yaml:44]: we likely wanna use anyOf phase0 and electra attestation to be able to switch to the api pre-electra

File: apis/validator/aggregate_attestation.v2.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  responses:
+    "200":
+      description: "Returns aggregated `Attestation` object with same `AttestationData` root, slot and committee index."
+      content:
+        application/json:
+          schema:
+            title: GetAggregatedAttestationV2Response
+            type: object
+            required: [data]
+            properties:
```
Comment (nflaig) [inline apis/validator/aggregate_attestation.v2.yaml:42]: let's make this api fork aware by adding version meta

**Comment by rkapka** [comment]
Adding to @nflaig 's comments, if we make committee index required then this endpoint will not be backwards compatible and validator clients will have to be aware of both.

**Review by nflaig — DISMISSED** [review]

File: types/electra/attestation.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Electra:
+  Attestation:
+    type: object
+    description: "The [`Attestation`](https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.2/specs/electra/beacon-chain.md#attestation) object from the CL spec."
```
Comment (nflaig) [inline types/electra/attestation.yaml:4]:
```suggestion
    description: "The [`Attestation`](https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.3/specs/electra/beacon-chain.md#attestation) object from the CL spec."
```

**Review by nflaig — APPROVED** [review]

File: types/electra/attestation.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  Attestation:
+    type: object
+    description: "The [`Attestation`](https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.3/specs/electra/beacon-chain.md#attestation) object from the CL spec."
+    required: [aggregation_bits, data, signature, committee_bits]
+    properties:
+      aggregation_bits:
+        $ref: "../primitive.yaml#/BitList"
+        description: "Attester aggregation bits."
+      data:
+        $ref: "../attestation.yaml#/AttestationData"
```
Comment (rkapka) [inline types/electra/attestation.yaml:11]: Since you added a description to every other field, can you add one to `data` too?

**Review by rkapka — APPROVED** [review]

File: types/electra/attestation.yaml (hunk 4 above)
Comment (nflaig) [reply types/electra/attestation.yaml:11]: The default description of `AttestationData` should be sufficient as it has not been modified in electra

https://github.com/ethereum/beacon-APIs/blob/672e03e25ace85a3bacaea553fbf374f4f844435/types/attestation.yaml#L46-L48

**Review by tersec — APPROVED** [review]

File: types/electra/attestation.yaml (hunk 4 above)
Comment (rolfyone) [reply types/electra/attestation.yaml:11]: I'm happy with that default description...

**Review by rolfyone — APPROVED** [review]

File: apis/validator/aggregate_attestation.v2.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      schema:
+        $ref: '../../beacon-node-oapi.yaml#/components/schemas/Uint64'
+    - name: committee_index
+      in: query
+      required: true
+      schema:
+        $ref: '../../beacon-node-oapi.yaml#/components/schemas/Uint64'
+  responses:
+    "200":
+      description: "Returns aggregated `Attestation` object with same `AttestationData` root, slot and committee index."
```
Comment (nflaig) [inline apis/validator/aggregate_attestation.v2.yaml:35]: Noticed the description will not be correct post-electra as `index` in `AttestationData` will be set to 0 and not match the  value passed in `committee_index` query param.

Comment (ensi321) [reply apis/validator/aggregate_attestation.v2.yaml:35]: @nflaig good catch. Will update now

Comment (ensi321) [reply apis/validator/aggregate_attestation.v2.yaml:35]: > Noticed the description will not be correct post-electra as `index` in `AttestationData` will be set to 0 and not match the value passed in `committee_index` query param.

I think we can interpret this as:
```
Returns aggregated `Attestation` object with the same root and slot in `Attestation.data`, and same committee index in `Attestation`. 
```

I think this description still holds true. Just that the committee index is now in `Attestation` and not `AttestationData`

**Pull request merged**